ngrinder-core/src/main/java/net/grinder/SingleConsole.java

```diff
@@ -86,6 +88,9 @@
 
 	private final Condition eventSyncCondition = new Condition();
 	private ProcessReports[] processReports;
+
+	// It contains cached distribution files md5 checksum from each agents.
+	private List<Set<String>> agentCachedDistFilesMd5List = synchronizedList(new ArrayList<>());

```

Comment:
How about use Concurrent Collections rather than Synchronized Collections?
https://javarevisited.blogspot.com/2016/05/what-is-difference-between-synchronized.html